src/v/kafka/server/group_manager.cc

```diff
@@ -567,6 +582,42 @@ group_manager::leave_group(leave_group_request&& r) {
     }
 }
 
+ss::future<txn_offset_commit_response>
+group_manager::txn_offset_commit(txn_offset_commit_request&& r) {
+    if (!_catchup_lock.try_read_lock()) {
```

Comment:
I was thinking about this but then I noticed that `try_lock; release; lock` looks weird when you don't know the intent and used finally to release the lock. Why do you think switching to RAII have benefits there?

Reply:
yep `try_lock; release; lock` does look weird!! it's a hack to work around the proper api not existing in seastar, but that isn't a limitation other than someone just needs to write the few lines of code and send a patch upstream.

switching to RAII isn't about a benefit to semantics. that is, your code looks completely correct. its more about using abstractions that tend to be safe by default. RAII prevents common mistakes like forgetting to unlock, or unlocking in a `then` instead of a `finally`. it's not a big deal either way, but RAII should be preferred when possible.

Reply:
rewrote it to RAII